circuit/src/dynamic_sets/attest_native.rs

```diff
@@ -0,0 +1,292 @@
+use crate::circuit::PoseidonNativeSponge;
+
+use halo2::halo2curves::{bn256::Fr, ff::PrimeField};
+use num_rational::BigRational;
+use secp256k1::ecdsa;
+use sha3::{Digest, Keccak256};
+use std::collections::HashMap;
+
+use super::native::SignedAttestation;
+
+/// ECDSA public key
+pub type ECDSAPublicKey = secp256k1::PublicKey;
+/// ECDSA signature
+pub type ECDSASignature = ecdsa::RecoverableSignature;
+
+fn keccak256(data: &[u8]) -> Vec<u8> {
+	let mut hasher = Keccak256::new();
+	hasher.update(data);
+	hasher.finalize().to_vec()
+}
+
+fn recover_ethereum_address_from_pk(public_key: ECDSAPublicKey) -> Fr {
+	let public_key_bytes = public_key.serialize_uncompressed();
+	let hashed_public_key = keccak256(&public_key_bytes[1..]);
+	let address_bytes = &hashed_public_key[hashed_public_key.len() - 20..];
+
+	let mut address_bytes_array = [0u8; 32];
+	address_bytes_array[..address_bytes.len()].copy_from_slice(address_bytes);
+
+	Fr::from_bytes(&address_bytes_array).unwrap()
+}
+
+/// Dynamic set for EigenTrust
+pub struct EigenTrustAttestationSet<
+	const NUM_NEIGHBOURS: usize,
+	const NUM_ITERATIONS: usize,
+	const INITIAL_SCORE: u128,
+> {
+	set: Vec<(Option<Fr>, Fr)>,
```

Comment:
We should not use `Option`, instead we should use Fr::zero(), for null/default values